app/modules/asset_groups/models.py

```diff
@@ -438,16 +436,20 @@ def detected(self, job_id, data):
         self.job_complete(str(job_id))
 
     # Used to build the response to AssetGroupSighting GET
-    def assets(self):
+    def get_assets(self):
         from app.modules.assets.schemas import DetailedAssetSchema
 
         asset_schema = DetailedAssetSchema(
             many=False, only=('guid', 'filename', 'src', 'annotations')
         )
-        resp = []
+
+        assets = {}
         for filename in self.config.get('assetReferences'):
             asset = self.asset_group.get_asset_for_file(filename)
             assert asset
+            assets[asset.guid] = asset
```

Comment:
Does this code work correctly?  `assets` is a dict `asset.guid -> asset` and:

```python
for asset in sorted(assets):
    json_msg, err = asset_schema.dump(asset)  # <- isn't "asset" actually the asset guid?
```

You can just do this instead:

```python
assets = []
for ...:
    assets.append(asset)

for asset in sorted(assets, key=lambda a: a.guid):
   ...
```

Reply:
That bit almost certainly doesn't work correctly. Thanks for that and thanks for the lambda key suggestion.